### DISCUSSION TITLE
Background on imageWithLogo

### DISCUSSION THREAD

**Discussion by oehoe** [chillerlan/php-qrcode#122] [Q&A] [answered]
Thanks for this great project! I'm using the imageWithLogo example and I end up with a background color '#f5f5f5'. The 'imageTransparent' and 'imageTransparencyBG' QROptions don't seem to work. How do I get these options working?

![pngbase](https://user-images.githubusercontent.com/9802555/160566993-0c7dd008-d6d7-4a11-9b98-ed70119011b0.png)

**Comment by codemasher** [comment]
Hi! I've added a bit of documentation re this in #121 - [see this comment for more info](https://github.com/chillerlan/php-qrcode/discussions/121#discussioncomment-2393164). Hope this helps!

**Reply by oehoe** [reply]
Hey, Thank you I read the information. What I got out of it is that the imageTransparencyBG option would set a different background color with imageTransparent false.

In my case the imageTransparent true does not render the image transparent. With transparent false, imageTransparencyBG did not change the background color.

Sorry if I missed the solution in your post.

**Comment by codemasher** [comment, marked as answer]
Ok, i think I've found the culprit. I've [added a line](https://github.com/chillerlan/php-qrcode/blob/585c328d9e8c8dc79e52cf7242f2e6eb3b4b91af/src/Output/QRGdImage.php#L91-L95) to ensure the circular dots are actually round on low scales (scale <= 20) - internally it scales the image up, draws the dots and scales it back to the expected size - all while using the `IMG_BICUBIC` filter - which is what messes with the background color. So try to set `QROptions::$scale` to a value greater than 20 or set `QROptions::$drawCircularModules` to `false`. 
Alternatively i would recommend to use the SVG output for more crisp circles and colors. :)

**Reply by codemasher** [reply]
I've added a bit of info [to the corresponding docblock](https://github.com/chillerlan/php-qrcode/commit/9a0db7feeedf514fbf4d9b1970c9bd2311b40d8c).

**Reply by oehoe** [reply]
Nice! I'm gonna give it a try! Thanks for looking into it.

**Reply by oehoe** [reply]
I set the scale to 25. Works!

**Reply by codemasher** [reply]
That's great! Thank you for the feedback!
